packages/@lwc/template-compiler/src/codegen/static-element-serializer.ts

```diff
@@ -21,6 +21,14 @@ const rawContentElements = new Set([
     'PLAINTEXT',
 ]);
 
+/**
+ * Escape all the characters that could break a JavaScript template string literal: "`" (backtick),
+ * "${" (dollar + open curly).
+ */
+function templateStringEscape(str: string): string {
+    return str.replace(/`/g, '\\`').replace(/\$\{/g, '\\${');
```

Comment:
Unfortunately this doesn't handle cases with the `\` in the HTML:

```
<div>Escape \`me`</div>
```

I looked around for some battle-tested template-string escaper, but the closest I found is [this](https://github.com/rameshvarun/escape-template-string/blob/8cd5dcfbd1ea0a091758750d8448e6e7220ecaf7/index.js#L2). Based on this, we could modify the code like:

```suggestion
    return str.replace(/\\/g, '\\\\').replace(/`/g, '\\`').replace(/\$\{/g, '\\${');
```

Even this makes me a bit nervous. I'm worried there is some case here that we're not able to think of, where the user is not only able to break the JS, but maybe even escape the sandbox and execute arbitrary JS inside of the `${}`.

Reply:
This is a really good catch. As a side effect it also updated other fixtures, which I think makes sense. That said I am not 100% certain of this. Could you also double-check @nolanlawson it makes sense for you as well?

Reply:
I looked at the snapshot changes and they look correct to me – it seems we weren't properly escaping backslashes before.

However, the only way to really know for sure is to add a Karma test IMO. Otherwise we have to rely on our squishy human brains to compile the code and make sure it's correct. 🙂